sdl_android/src/main/java/com/smartdevicelink/api/screen/SoftButtonObject.java

```diff
@@ -0,0 +1,261 @@
+package com.smartdevicelink.api.screen;
+
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import com.smartdevicelink.proxy.rpc.OnButtonEvent;
+import com.smartdevicelink.proxy.rpc.OnButtonPress;
+import com.smartdevicelink.proxy.rpc.SoftButton;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * <strong>SoftButtonObject</strong> <br>
+ * SoftButtonObject define a button that can have multiple SoftButtonState values.<br>
+ * The states of SoftButtonObject allow the developer to not have to manage multiple SoftButtons that have very similar functionality.<br>
+ * For example, a repeat button in a music app can be thought of as one SoftButtonObject with three typical states: repeat off, repeat 1, and repeat on.<br>
+ * @see SoftButtonState
+ */
+public class SoftButtonObject {
+
+    private static final String TAG = "SoftButtonObject";
+    private String name;
+    private List<SoftButtonState> states;
+    private String currentStateName;
+    private int buttonId;
+    private OnEventListener onEventListener;
+    private SoftButtonManager softButtonManager;
```

Comment:
This seems excessive for a single method called and creates a circular logic. I would propose introducing an `UpdateListener` interface in this object, making a a `final` instantiation in the SoftButtonManager and passing that into each object instead.

Reply:
Agree. I made the changes. Let me know if the naming is good or need to be changed.